### ISSUE TITLE
Support Python 3 and IPython and add more examples

### ISSUE THREAD

**Issue by ErmiaAzarkhalili** [maxpumperla/hyperas#44]
Hi Max,
I open this issue to officially and kindly ask you to add python 3 and IPython support for both hyperapt and hyperas since you can maintain hyperapt now.

I tried many other packages and wrappers to tune keras, but your package is significantly more efficient than others.
Thanks in advance.

**Comment by MannyKayy** [comment]
Check out my [fork](https://github.com/MannyKayy/hyperas). Doesn't support Jupyter notebooks yet but should work with Python 3.

**Comment by ErmiaAzarkhalili** [comment]
Could you modify it to support jupyter?

Sincerely,
Ermia
*------------------------------------------------------------------------------------------------------------*
Never lose the child-like wonder. It's just too important. It's what drives
us.

**Comment by maxpumperla** [comment]
@ErmiaAzarkhalili 

- python3 is fixed
- what you describe as ipython is actually supporting `pycharm`. Please install master and see it it works now.
- Well, more examples is always cool, but it's not an actual issue. There's about 5 examples now, should just get you started.

I think I'll close this ticket for now. If you still have problems, please just reopen another one. Just make sure not to lump to many things in a ticket. One issue per issue, please. 👍

**Comment by ErmiaAzarkhalili** [comment]
Thanks a lot. @maxpumperla will obey your instructions...

**Comment by ErmiaAzarkhalili** [comment]
@maxpumperla I am still confused how I should implement my code in pure pycharm?
`from __future__ import print_function
from hyperopt import Trials, STATUS_OK, tpe
from hyperas import optim
from hyperas.distributions import choice, uniform

from keras.models import Sequential
from keras.layers.core import Dense, Dropout, Activation
from keras.optimizers import RMSprop

from keras.datasets import mnist
from keras.utils import np_utils


def data():
    """
    Data providing function:

    This function is separated from model() so that hyperopt
    won't reload data for each evaluation run.
    """
    (X_train, y_train), (X_test, y_test) = mnist.load_data()
    X_train = X_train.reshape(60000, 784)
    X_test = X_test.reshape(10000, 784)
    X_train = X_train.astype('float32')
    X_test = X_test.astype('float32')
    X_train /= 255
    X_test /= 255
    nb_classes = 10
    Y_train = np_utils.to_categorical(y_train, nb_classes)
    Y_test = np_utils.to_categorical(y_test, nb_classes)
    return X_train, Y_train, X_test, Y_test


def model(X_train, Y_train, X_test, Y_test):
    """
    Model providing function:

    Create Keras model with double curly brackets dropped-in as needed.
    Return value has to be a valid python dictionary with two customary keys:
        - loss: Specify a numeric evaluation metric to be minimized
        - status: Just use STATUS_OK and see hyperopt documentation if not feasible
    The last one is optional, though recommended, namely:
        - model: specify the model just created so that we can later use it again.
    """
    model = Sequential()
    model.add(Dense(512, input_shape=(784,)))
    model.add(Activation('relu'))
    model.add(Dropout({{uniform(0, 1)}}))
    model.add(Dense({{choice([256, 512, 1024])}}))
    model.add(Activation('relu'))
    model.add(Dropout({{uniform(0, 1)}}))
    model.add(Dense(10))
    model.add(Activation('softmax'))

    rms = RMSprop()
    model.compile(loss='categorical_crossentropy', optimizer=rms, metrics=['accuracy'])

    model.fit(X_train, Y_train,
              batch_size={{choice([64, 128])}},
              nb_epoch=1,
              verbose=2,
              validation_data=(X_test, Y_test))
    score, acc = model.evaluate(X_test, Y_test, verbose=0)
    print('Test accuracy:', acc)
    return {'loss': -acc, 'status': STATUS_OK, 'model': model}


X_train, Y_train, X_test, Y_test = data()

best_run, best_model = optim.minimize(model=model,
                                      data=data,
                                      algo=tpe.suggest,
                                      max_evals=5,
                                      trials=Trials(), notebook_name="simple_notebook")
best_model.save("best_model.h5")
print("Evalutation of best performing model:")
print(best_model.evaluate(X_test, Y_test))
`
it's my code. Should I crate any notebook file? 
1) I create and empty it. then run my code and get the error. "mnist is not found"
2) then I copy-paste my code into notebook, and run my code. again get the error "mnist is not found"
3) I run the notebook directly, everything is OK. then run pycharm file and it is OK as well.
4) empty the notebook, and run pycharm code.. it is still surprisingly working... 

I would be grateful if you could provide me as to how utilize hyperas in pure pycharm code...
